clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala

```diff
@@ -153,9 +154,12 @@ object UncommittedGarbageCollector {
       removed = {
         if (shouldSweep) {
           if (markID != "") { // get the expired addresses from the mark id run
-            addressesToDelete = readMarkedAddresses(storageNamespace, markID)
+            markedAddresses = readMarkedAddresses(storageNamespace, markID)
```

Comment:
There's a bug here, if we run mark&sweep and pass a markID we will not deleted the items we marked.
We should assert that there's no MarkID if mark == true

Reply:
Also it's better to change the conditions to: if mark==true

Reply:
Hi, it’s not a bug, if we pass a mark id it will delete it even if we also mark in this run
It's the same as the GC behavior

Reply:
We are handling the addresses to delete differently from how the committed GC is working. I don't see any valid use case for passing mark ID with mark == true and I feel it might cause more confusion for us and the users. In what kind of scenario do you see the user pass both markID and performs mark and sweep?

Reply:
We discussed it and decided to block the option to the use to pass mark id when performing mark